frontend/src/modules/accounts/components/AccountsTable.tsx

```diff
@@ -0,0 +1,163 @@
+import {
+  Avatar,
+  Flex,
+  HStack,
+  Icon,
+  Table,
+  TableContainer,
+  Tag,
+  Tbody,
+  Td,
+  Text,
+  Th,
+  Thead,
+  Tr,
+  useDisclosure,
+} from '@chakra-ui/react';
+import { UserDTO } from 'generated-api';
+import { useRouter } from 'next/router';
+import { useContext, useEffect, useRef, useState } from 'react';
+import { FaHandPointRight, FaLock, FaUserAlt } from 'react-icons/fa';
+import { useMutation } from 'react-query';
+import { Dialog } from '../../../shared/components/elements/Dialog/Dialog';
+import { useToast } from '../../../shared/hooks';
+import { ApiContext } from '../../../shared/providers/ApiProvider';
+import { useGlobalStore } from '../../../shared/stores';
+import { AccountActions } from './AccountActions';
+
+export const AccountsTable = () => {
+  const api = useContext(ApiContext);
+  const router = useRouter();
+  const toast = useToast();
+
+  const { getUser } = useGlobalStore();
+  const localUser = getUser();
+
+  const { isOpen, onOpen, onClose } = useDisclosure();
+  const [users, setUsers] = useState<UserDTO[] | undefined>([]);
+  const userToDelete = useRef<UserDTO | null>(null);
+
+  const fetchUsers = useMutation(() => api.getUsers(), {
+    onSuccess: (data) => {
+      setUsers(() => data.data);
+    },
+  });
```

Comment:
for "get" operations, we want to use useQuery instead of useMutation as it gives us caching automatically. useMutation should only be used for create/read/edit queries